saml/saml.php

```diff
@@ -0,0 +1,429 @@
+<?php
+/*
+ * Name: SAML SSO and SLO
+ * Description: replace login and registration with a SAML identity provider.
+ * Version: 0.0
+ * Author: Ryan <https://friendica.verya.pe/profile/ryan>
+ */
+use Friendica\Core\Hook;
+use Friendica\Core\Logger;
+use Friendica\Core\Renderer;
+use Friendica\Core\Session;
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Model\User;
+use Friendica\Util\Strings;
+use OneLogin\Saml2\Auth;
+
+require_once(__DIR__ . '/vendor/autoload.php');
+
+define("PW_LEN", 32); // number of characters to use for random passwords
+
+function saml_module($a) {}
+
+function saml_init($a) {
```

Comment:
Code standards: We use a very slight variation of the PSR-2 coding standards that mandates a new line before function opening braces among other things, please check the rest of the basic rules: https://friendica.mrpetovan.com/help/Developers-Intro#Coding+standards